scripts/atomic_commit_manager.py

```
@@ -0,0 +1,269 @@
+            
+        if group.status != "failed":
+            print(f"Commit group {group_id} is not in failed status")
+            return False
+            
+        try:
+            # Rollback each worktree that had a successful commit
+            for worktree, commit_hash in group.commit_hashes.items():
+                worktree_path = self.project_root / "worktrees" / worktree
+                if worktree_path.exists() and commit_hash:
+                    # Reset to previous commit
+                    subprocess.run([
```

Ruff (0.14.3)

`subprocess` call: check for execution of untrusted input

(S603)